cmake/EnableCompilerWarnings.cmake

```diff
@@ -96,6 +96,7 @@ else()
                 -Wno-unsafe-buffer-usage
                 -Wno-deprecated-declarations
                 -Wno-shadow-uncaptured-local
+                -Wno-return-type
```

Comment:
@atamazov @JehandadKhan @DrizztDoUrden @averinevg @dmikushin 
This line is due to compiler is not parsing the following lines well:
https://github.com/ROCmSoftwarePlatform/MIOpen/blob/b73554749188f12984b3acb776bdc065ca1d20ec/src/include/miopen/errors.hpp#L60-L64

Since `MIOPEN_THROW` is used in so many different places, it's challenging to suppress them individually. 
I believe this is a false positive of the compiler being thorough but not too clever, we have `throw` in the above lines but its static check still doubts some paths do not have proper returns.

Reply:
We could try wrapping its content in a function and adding [[noreturn]]. I didn't try, but theoretically should work.

Reply:
I mean an approach like this:
```C++
template <class... Params>
[[noreturn]] void MiopenThrow(const std::string& file, int line, Params&&... args)
{
	throw miopen::Exception(std::forward<Params>(args)...).SetContext(file, line);
}

#define MIOPEN_THROW(...) MiopenThrow(__FILE__, __LINE__, __VA_ARGS__)
```